security/user-last-seen.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+const LAST_SEEN_UPDATE_USER_META_CACHE_TTL = 60; // Store last seen once per minute to avoid too much write DB operations
+const LAST_SEEN_UPDATE_USER_META_CACHE_KEY_PREFIX = 'vip_last_seen_update_user_meta_cache_key';
+
+add_action( 'set_current_user', 'update_user_last_seen', 10, 0 );
+add_filter('manage_users_columns', 'users_columns' ) ;
+add_filter('manage_users_custom_column', 'users_custom_column', 10, 3 ) ;
+
+function update_user_last_seen() {
+	global $current_user;
+
+	$cache_key = LAST_SEEN_UPDATE_USER_META_CACHE_KEY_PREFIX . $current_user->ID;
+
+	try {
+		if ( wp_cache_get( $cache_key ) ) {
+			return;
+		}
+
+		update_user_meta( $current_user->ID, 'vip_last_seen', time() );
+
+		wp_cache_set( $cache_key, true, null, LAST_SEEN_UPDATE_USER_META_CACHE_TTL );
```

Comment:
This would be less race-condition prone:

```
if ( wp_cache_add( $cache_key, true, 'vip', LAST_SEEN_UPDATE_USER_META_CACHE_TTL ) ) {
	update_user_meta( $current_user->ID, 'vip_last_seen', time() );
}
```